### PR TITLE
Feature/issue 3095 test role access changes

### PR DESCRIPTION
**Pull request by tleyden**
# Integration tests

Single test only

http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/230/
http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/231/

Full suite

http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/232/
http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/233/

### CODE REVIEW

**Comment by tleyden** [comment]
Fixes #3095 TestRoleAccessChanges sporadic failure by converting changes requests to rt.WaitForChanges().

I was unable to reproduce the error, so this is a “blind fix”

File: rest/api_test.go [hunk 1]
```diff
@@ -2157,29 +2157,41 @@ func TestRoleAccessChanges(t *testing.T) {
...
-	var changes struct {
-		Results  []db.ChangeEntry
-		Last_Seq interface{}
-	}
-	response = rt.Send(requestByUser("GET", "/db/_changes", "", "alice"))
-	log.Printf("1st _changes looks like: %s", response.Body.Bytes())
-	json.Unmarshal(response.Body.Bytes(), &changes)
-	assert.Equals(t, len(changes.Results), 2)
-	since := changes.Last_Seq
+	changesResult, err := rt.WaitForChanges(2, "/db/_changes", "alice", false)
```
Comment (adamcfraser) [inline rest/api_test.go]: It's probably fine for this test, but in general WaitForSequence feels like a better choice than WaitForChanges, as there's less impact on SG state.  (e.g. the authentication on each changes request could modify the user document,  and so WaitForChanges would hide an issue where the auth is expecteded to work the first time, but instead only works on the second time).

**Review by adamcfraser — APPROVED** [review]
General feedback about WaitForSequence vs. WaitForChanges, but doesn't need to block this fix.

**Comment by tleyden** [comment]
I pushed up an update version that uses `rt.WaitForSequence()` instead

File: rest/utilities_testing.go [hunk 2]
```diff
@@ -244,7 +244,8 @@ func (rt *RestTester) TestPublicHandler() http.Handler {
 }
 
 type changesResults struct {
-	Results []db.ChangeEntry
+	Results  []db.ChangeEntry
+	Last_Seq interface{}
```
Comment (tleyden) [inline rest/utilities_testing.go:249]: Not strictly needed as part of this PR, but will likely be needed in future usage of `rt.WaitForChanges()`

**Comment by tleyden** [comment]
@adamcfraser can you review latest commits and then merge?

**Pull request merged**